src/pgduckdb_types.cpp

```diff
@@ -301,7 +540,6 @@ ConvertDuckToPostgresArray(TupleTableSlot *slot, duckdb::Value &value, idx_t col
 	D_ASSERT(value.type().id() == duckdb::LogicalTypeId::LIST);
 	auto &child_type = GetChildTypeRecursive(value.type());
 	auto child_id = child_type.id();
-	D_ASSERT(child_id == OP::ExpectedType());
```

Comment:
Since you've removed the `D_ASSERT`, you can now remove the unused variable `child_id`?